### DISCUSSION TITLE
zh/guide/install/script

### DISCUSSION THREAD

**Discussion by giscus** [AlistGo/docs#27] [Announcements]
# zh/guide/install/script

AList v3的文档

https://alist.nn.ci/zh/guide/install/script.html

**Comment by jinchunyou** [comment]
curl performs SSL certificate verification by default, using a "bundle"
 of Certificate Authority (CA) public keys (CA certs). If the default
 bundle file isn't adequate, you can specify an alternate file
 using the --cacert option.
If this HTTPS server uses a certificate signed by a CA represented in
 the bundle, the certificate verification probably failed due to a
 problem with the certificate (it might be expired, or the name might
 not match the domain name in the URL).
If you'd like to turn off curl's verification of the certificate, use
 the -k (or --insecure) option.


请问有解决办法吗

**Reply by tesly03** [reply]
你可以删除https中的s

**Reply by naponorn** [reply]
太复杂了

**Comment by minshurui** [comment]
新版本能不能出个备份还原的功能啊

**Reply by outt-iao** [reply]
备份/alist/data/data.db

**Reply by wyf9** [reply]
现在已经有备份还原配置的功能了，在管理后台里

**Comment by chubbyginger** [comment]
linux一键安装脚本失败
```
tar (child): /tmp/alist.tar.gz：无法 open: 没有那个文件或目录
tar (child): Error is not recoverable: exiting now
tar: Child returned status 2
tar: Error is not recoverable: exiting now
下载 alist-linux-musl-amd64.tar.gz 失败！
```

**Reply by H0wAway** [reply]
一样

**Reply by dalaoshi777** [reply]
解压参数貌似有问题

**Reply by H0wAway** [reply]
无所谓，手动安装也简单

**Comment by zgswq** [comment]
下载 Alist latest ...
####################################################################################################################################################################################################################################### 100.0%

gzip: stdin: not in gzip format
tar: Child returned status 1
tar: Error is not recoverable: exiting now
下载 alist-linux-musl-amd64.tar.gz 失败！

**Comment by andywang1011** [comment]
默认管理账号密码呢

**Reply by zgswq** [reply]
昨天提交问题2分钟后，上传了文件，这个问题就解决了。文件找不到引发的错误。 ![image](https://user-images.githubusercontent.com/77392265/195541214-17bc8548-fe23-494a-a34e-b5901c571084.png)

**Comment by henryking1380** [comment]
failed get storage: can't find storage with rawPath: /
请问主页显示怎么解决，谢谢

**Reply by zgswq** [reply]
看官方文档：https://alist.nn.ci/zh/faq/why.html

**Reply by henryking1380** [reply]
谢谢！

**Comment by mickey0891** [comment]
请问windows10如何卸载从新安装？

**Comment by qian9008** [comment]
今天突然白屏了

**Comment by nx9000** [comment]
请问提示curl: (6) Couldn't resolve host 'alist.nn.ci'，怎么办？

**Comment by yyddtj** [comment]
一键脚本有开机自启吗

**Reply by zorgzola** [reply]
有的。

**Comment by zorgzola** [comment]
一键脚本安装好后，获取admin的密码后，登录密码错误，怎么解决？
谢谢。

**Comment by bincooo** [comment]
v3.4.0
alist admin的密码一直提示错误password is incorrect

**Reply by bincooo** [reply]
我输入的密码是alist admin指令结果中复制的

**Reply by zorgzola** [reply]
安装手动版没问题，可以运行一次程序后复制config和data文件。然后安装一键脚本，之前复制的文件覆盖过去就可以了。我这样解决了。

**Reply by andywang1011** [reply]
谢谢，找解决了，但是也谢谢你的回答。

**Comment by leoneoo** [comment]
curl -fsSL "https://alist.nn.ci/v3.sh" | bash -s update  使用一键更新脚本，无法更新到3.4.0版本。请问该如何正确升级呢？

**Comment by bjhwjy** [comment]
![](https://github.com/bjhwjy/blog-img/202212082343838.jpg)

**Comment by bjhwjy** [comment]
![](https://github.com/bjhwjy/blog-img/202212082343838.jpg)
请问这种需要怎么解决呢？

**Comment by bjhwjy** [comment]
(https://github.com/bjhwjy/blog-img/202212082343838.jpg)

**Comment by Dough-su** [comment]
System error: SyntaxError: Invalid regular expression: /?/: Nothing to repeat

**Comment by Hi-360** [comment]
你好  我想问下 centos7或者centos8能安装吗？x86平台的

**Reply by 1203952894** [reply]
目前我是用 centos7 中通过 docker 成功安装

**Comment by ssdswi** [comment]
如何知道自己所安装的alist版本

**Comment by lpzl636** [comment]
搭建好后搜索的时候报错：search not available
这是什么原因？

**Comment by FopFat** [comment]
有没有什么办法可以让Alist服务定期重启？

**Comment by e5shujufenxifenxi** [comment]
你好，我已经把程序安装到腾讯云网站上了，但是我不想让资源页面被所有人公开访问，我仅仅是想自己的几十个设备私密访问，请问我怎么设置才能对资源页面进行加密，需要凭借账号密码才能访问，或者对资源页面进行隐藏，不登录账号密码就不允许看见。请问要怎么设置才能实现这样的目的？？

**Comment by 1203952894** [comment]
linux服务器下可使用pm2将指定端口进行重启，详细请参考，http://43.143.184.147/zh/Study/Skills/skills02.html



---Original---
From: ***@***.***&gt;
Date: Sun, Jan 15, 2023 01:39 AM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [alist-org/docs] zh/guide/install/script (Discussion #27)




 
有没有什么办法可以让Alist服务定期重启？

**Comment by GUO-tongxue** [comment]
下载 Alist latest ...
##############################                                            42.0%
curl: (18) transfer closed with 8162251 bytes remaining to read

gzip: stdin: unexpected end of file
tar: 归档文件中异常的 EOF
tar: 归档文件中异常的 EOF
tar: Error is not recoverable: exiting now
 下载成功 
---------管理员信息--------
bash: 行 171: 16631 段错误               ./alist admin
--------------------------

**Comment by STEVENTAN100** [comment]
3.10.1的链接脚本没有更新

**Comment by STEVENTAN100** [comment]
一键脚本没有更新到3.11.0

**Comment by 84430701** [comment]
如何一个服务器搭建多个alist网盘

**Comment by Salomens** [comment]
我使用脚本一键安装了alist，但是今天早上发现所有服务都崩掉了。我就用那个卸载命令删除了他，可是我的存储空间却仍然有100多m的空间占用，这些没有删除完全的文件我该在哪里删除它

**Comment by ladychili** [comment]
Padavan 用 ssh admin进入之后用一键脚本安装，报错说“出错了，请使用 root 权限重试！”怎么办啊 .. padavan只有admin/admin

**Reply by UareoVo** [reply]
su - root 
切换到root用户安装呗

**Comment by UareoVo** [comment]
windows 可以在cmd 下通过winget 安装（winget install alist） 更新（winget upgrade alist）卸载(winget uninstall alist) 很快 很方便  不支持winget命令的可以去微软应用商店搜索“软件包管理器” 安装一下

**Comment by ZIDC** [comment]
alist.service - Alist service
Loaded: loaded (/etc/systemd/system/a>
Active: failed (Result: exit-code) si
Process: 419550 ExecStart=/www/wwwroot
Main PID: 419550 (code=exited, status=1
这怎么解决哎

**Reply by ZIDC** [reply]
路径问题，已经解决

**Comment by axixa120** [comment]
出错了，请使用 root 权限重试！ 这是怎么回事

**Reply by bian2022** [reply]
看不懂吗？
用root账号

**Comment by aszhangshuai** [comment]
更新提示：curl: (7) Failed connect to alist.nn.ci:443; No route to host
怎么回事这是？

**Comment by listenwindow** [comment]
[root@localhost ~]# curl -fsSL "https://alist.nn.ci/v3.sh" | bash -s update
curl: (6) Could not resolve host: alist.nn.ci; 未知的错误

**Comment by listenwindow** [comment]
[root@localhost ~]# curl -fsSL "https://alist.nn.ci/v3.sh" | bash -s update
curl: (6) Could not resolve host: alist.nn.ci; 未知的错误

**Comment by kjmnxcfdg** [comment]
无法用一键脚本更新了，以前同一个命令都可以，现在提示出错了，当前系统未安装 Alist。备份了目录里面的config.json和data.db，用重装命令，提示此位置已经安装，请选择其他位置，或使用更新命令。自己手动下载了最新版alist-linux-musl-amd64.tar.gz解压替换进去，重启，网站报错502 Bad Gateway
检查状态提示
● alist.service - Alist service
   Loaded: loaded (/etc/systemd/system/alist.service; enabled; vendor preset: disabled)
   Active: active (running) since Tue 2023-08-08 09:35:52 CST; 12s ago
 Main PID: 26010 (alist)
   CGroup: /system.slice/alist.service
           └─26010 /www/wwwroot/**********/alistv3/alist/alist server
Aug 08 09:35:52 en-cvm-3k7t5i623i224.novalocal systemd[1]: Started Alist service.
Aug 08 09:35:52 en-cvm-3k7t5i623i224.novalocal alist[26010]: INFO[2023-08-08 09:35:52] reading config fil...on
Aug 08 09:35:52 en-cvm-3k7t5i623i224.novalocal alist[26010]: INFO[2023-08-08 09:35:52] load config from e...T_
Aug 08 09:35:52 en-cvm-3k7t5i623i224.novalocal alist[26010]: INFO[2023-08-08 09:35:52] init logrus...
Aug 08 09:35:52 en-cvm-3k7t5i623i224.novalocal alist[26010]: INFO[2023-08-08 09:35:52] start HTTP server ...44
Aug 08 09:35:52 en-cvm-3k7t5i623i224.novalocal alist[26010]: INFO[2023-08-08 09:35:52] qbittorrent not ready.
Aug 08 09:35:52 en-cvm-3k7t5i623i224.novalocal alist[26010]: INFO[2023-08-08 09:35:52] Aria2 not ready.
Hint: Some lines were ellipsized, use -l to show in full.

**Reply by kjmnxcfdg** [reply]
前面的问题找到原因了，命令的目录输入错误了。
更新后不能打开网页也找到原因了，配置文件会强行更改端口到默认，需要自己改回来。

**Reply by STEVENTAN100** [reply]
你可以使用caddy或nginx反向代理，alist配置文件使用默认的，监听127.0.0.1，这样就不用每次更新就搞一回了

**Comment by sz173510443** [comment]
使用一键脚本更新后，网页一直卡在获取设置失败： 请稍候，正在加载储存

**Comment by yipianxiaoye** [comment]
哪位大佬帮我看下怎么回事
下载 Alist latest ...
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0curl: (7) Couldn't connect to server
tar (child): /tmp/alist.tar.gz: Cannot open: No such file or directory
tar (child): Error is not recoverable: exiting now
tar: Child returned status 2
tar: Error is not recoverable: exiting now
下载 alist-linux-musl-amd64.tar.gz 失败！

**Comment by ziyou8513** [comment]
一键安装alist以后，服务器上的443端口无法处理请求，重启后解决，这是什么原因？

**Comment by bian2022** [comment]
一键脚本能不能支持下Alpine系统？

**Comment by ziyou8513** [comment]
是用root用户安装的发自我的iPhone------------------ Original ------------------From: wumingshi  ***@***.***>Date: Sun,Sep 10,2023 2:10 AMTo: alist-org/docs ***@***.***>Cc: duanxufei ***@***.***>, Comment ***@***.***>Subject: Re: [alist-org/docs] zh/guide/install/script (Discussion #27)
看不懂吗？
用root账号

—Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you commented.Message ID: ***@***.***>
[
{
***@***.***": "http://schema.org",
***@***.***": "EmailMessage",
"potentialAction": {
***@***.***": "ViewAction",
"target": "https://github.com/alist-org/docs/discussions/27#discussioncomment-6956653",
"url": "https://github.com/alist-org/docs/discussions/27#discussioncomment-6956653",
"name": "View Discussion"
},
"description": "View this Discussion on GitHub",
"publisher": {
***@***.***": "Organization",
"name": "GitHub",
"url": "https://github.com"
}
}
]

**Comment by meowhuan** [comment]
System error: TypeError: n.replaceAll is not a function
系统错误：TypeError:n.replaceAll不是函数
是浏览器的问题吗？我换了浏览器就没事

**Comment by lish1609** [comment]
这个怎么整啊，各种方法都试过了

 % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:--  0:02:10 --:--:--     0
curl: (28) Failed to connect to ghproxy.com port 443 after 130540 ms: Connection timed out
tar (child): /tmp/alist.tar.gz: Cannot open: No such file or directory
tar (child): Error is not recoverable: exiting now
tar: Child returned status 2
tar: Error is not recoverable: exiting now
下载 alist-linux-musl-amd64.tar.gz 失败！

**Comment by tanny77** [comment]
下载 Alist latest ...
-=O=-                                         #    #     #    #               
curl: (7) Failed to connect to github.abskoop.workers.dev port 443: Connection timed out
tar (child): /tmp/alist.tar.gz: Cannot open: No such file or directory
tar (child): Error is not recoverable: exiting now
tar: Child returned status 2
tar: Error is not recoverable: exiting now
下载 alist-linux-musl-amd64.tar.gz 失败！

**Comment by Qliu-xiaoguai** [comment]
密码一直错误，怎么解决啊，求解

**Comment by Qliu-xiaoguai** [comment]
密码一直错误，我是按照获取密码，然后在去登录，密码就是不对

**Reply by shine-fire** [reply]
直接重置密码试试吧

**Reply by Qliu-xiaoguai** [reply]
我明白了，是我cd的路径问题

**Comment by mosder66** [comment]
设置密码总是出现这个问题，也不知道默认密码是多少

WARN[2023-11-19 08:50:45] [del_user_cache_online] failed: Post "http://localhost:5244/api/admin/user/del_cache?username=admin": dial tcp [::1]:5244: connect: connection refused 
[

**Comment by iwen-conf** [comment]
新手，比较好奇alist使用的是前后端分离还是前后端不分离啊

**Reply by BlueSkyXN** [reply]
源代码分离
二进制上集成

**Comment by chiquitita1227** [comment]
一键脚本安装失败啊，最后提示这个错误，有解决办法没有？

下载 alist-linux-musl-amd64.tar.gz 失败！

**Reply by BlueSkyXN** [reply]
> 一键脚本安装失败啊，最后提示这个错误，有解决办法没有？
> 
> 下载 alist-linux-musl-amd64.tar.gz 失败！

源的问题
使用 https://github.com/BlueSkyXN/ChangeSource/blob/master/alist.sh

**Comment by chunhanzi** [comment]
输入 curl -fsSL "https://alist.nn.ci/v3.sh" | bash -s install  后出现以下提示
curl -fsSL "https://alist.nn.ci/v3.sh" | bash -s install
请各位大佬指引

**Reply by BlueSkyXN** [reply]
6

**Reply by Jaffrez** [reply]
![006APoFYly1g5mu7no6ykj305i04gmwz](https://github.com/alist-org/docs/assets/117080567/6d225ac6-ab26-43aa-a428-cde8c54a572b)
我甘拜下风

**Comment by SalicylicAcid** [comment]
ubuntu服务器安装完一键脚本遇到503错误，页面打不开。

**Reply by SalicylicAcid** [reply]
破案了，iptables没关

**Comment by meowhuan** [comment]
6

猫羽雫
***@***.***








---Original---
From: ***@***.***&gt;
Date: Fri, Dec 15, 2023 20:05 PM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [alist-org/docs] zh/guide/install/script (Discussion #27)




 
破案了，iptables没关
 
&mdash;

**Comment by starsei** [comment]
# curl -fsSL "https://alist.nn.ci/v3.sh" | bash -s update
curl: (35) Encountered end of file

一键更新提示这样

**Comment by ReidLiao** [comment]
下载 alist-linux-musl-amd64.tar.gz 失败！

**Comment by hello-aiai** [comment]
今天更新alist后localhost:5244主页显示为空，localhost:5244/@manage管理界面显示和操作正常，是新版本的问题吗？

**Reply by zorgzola** [reply]
看下这里：
https://github.com/alist-org/alist/issues/6100

**Reply by hello-aiai** [reply]
Thanks！已解决。

**Comment by daqyt** [comment]
设置密码的时候出现一下问题怎么解决
root@ser921569246650 ~# /bot/alist2/alist admin random
INFO[2024-03-07 22:01:10] reading config file: data/config.json
INFO[2024-03-07 22:01:10] load config from env with prefix: ALIST_
INFO[2024-03-07 22:01:10] init logrus...
INFO[2024-03-07 22:01:10] admin user has been updated:
INFO[2024-03-07 22:01:10] username: admin
INFO[2024-03-07 22:01:10] password: IJ0guhr7
ERRO[2024-03-07 22:01:10] [del_user_cache_online] error: that's not even a token

**Comment by yuf19-2022** [comment]
安装完了这样：[root@iZwz9dul4nesaetcasjc9aZ alist]# systemctl status alist
● alist.service - Alist service
   Loaded: loaded (/etc/systemd/system/alist.service; enabled; vendor preset: disabled)
   Active: failed (Result: exit-code) since Sun 2024-03-31 09:27:06 CST; 5min ago
  Process: 27839 ExecStart=/usr/local/alist/alist server (code=exited, status=1/FAILURE)
 Main PID: 27839 (code=exited, status=1/FAILURE)

Mar 31 09:27:06 iZwz9dul4nesaetcasjc9aZ alist[27839]: INFO[2024-03-31 09:27:05] Successfully created the admin user and the initial password is: CcmeJY68
Mar 31 09:27:06 iZwz9dul4nesaetcasjc9aZ alist[27839]: WARN[2024-03-31 09:27:06] init tool aria2 failed: failed get aria2 version: Post "http://localhost:6800/jsonrpc": dial tcp...ion refused
Mar 31 09:27:06 iZwz9dul4nesaetcasjc9aZ alist[27839]: INFO[2024-03-31 09:27:06] init tool SimpleHttp success: ok
Mar 31 09:27:06 iZwz9dul4nesaetcasjc9aZ alist[27839]: WARN[2024-03-31 09:27:06] init tool qBittorrent failed: Post "http://localhost:8080/api/v2/auth/login": dial tcp [::1]:808...ion refused
Mar 31 09:27:06 iZwz9dul4nesaetcasjc9aZ alist[27839]: INFO[2024-03-31 09:27:06] start HTTP server @ 0.0.0.0:5244
Mar 31 09:27:06 iZwz9dul4nesaetcasjc9aZ alist[27839]: INFO[2024-03-31 09:27:06] start S3 server @ 0.0.0.0:5246
Mar 31 09:27:06 iZwz9dul4nesaetcasjc9aZ alist[27839]: FATA[2024-03-31 09:27:06] failed to start http: listen tcp 0.0.0.0:5246: bind: address already in use
Mar 31 09:27:06 iZwz9dul4nesaetcasjc9aZ systemd[1]: alist.service: main process exited, code=exited, status=1/FAILURE
Mar 31 09:27:06 iZwz9dul4nesaetcasjc9aZ systemd[1]: Unit alist.service entered failed state.
Mar 31 09:27:06 iZwz9dul4nesaetcasjc9aZ systemd[1]: alist.service failed.
Hint: Some lines were ellipsized, use -l to show in full.
[root@iZwz9dul4nesaetcasjc9aZ alist]#

**Comment by katfionn** [comment]
termux里面的alist怎么更新？想单独更新alist，不想全部包一起更

**Reply by wyf9** [reply]
替换可执行文件？
~~What time is it?~~

**Comment by zx108wl** [comment]
为什么，软件设计在运行路径生成data配置路径？我将alist软连接到了/usr/bin目录，在我的用户目录下直接运行alist他给我在我的运行目录下生成了data目录，这样太不方便管理了，请问又解决办法吗？,

**Reply by zorgzola** [reply]
都是在/usr/bin/alist 下，变通一下有那么难？

**Reply by wyf9** [reply]
如果不想移alist位置：启动前cd到别的目录即可

> (貌似放在 /opt/alist 也没多难管理？)
